### PR TITLE
Deprecate standalone kani, add support for cargo-kani for every call

### PR DESCRIPTION
**Pull request by jaisnan**
### Description of changes: 

Switch to cargo-kani for all commands, and deprecate usage of standalone kani.

### Call-outs:

Need to raise an issue or call out in documentation that we don't support standalone kani as a proof runner i.e we require that the extension be run on a crate with a declared cargo.toml. This is similar to how rust-analyzer handles it too, https://github.com/rust-lang/rust-analyzer/issues/8818.  

### Testing:

* How is this change tested? Manual testing

* Is this a refactor change? Yes

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
 	"compilerOptions": {
 		"module": "commonjs",
-		"target": "ES2020",
+		"target": "ES2018",
```
Comment (adpaco-aws) [inline tsconfig.json:4]: Why are these changes needed?

File: src/model/kaniCommandCreate.ts [hunk 2]
```diff
@@ -98,26 +74,14 @@ export async function captureFailedChecks(
...
-		let harnessCommand = '';
 
-		if (args === undefined || NaN) {
-			harnessCommand = `${KaniConstants.CargoKaniExecutableName} ${KaniArguments.harnessFlag} ${harnessName} --output-format terse`;
-		} else {
-			harnessCommand = `${KaniConstants.CargoKaniExecutableName} ${KaniArguments.harnessFlag} ${harnessName} ${KaniArguments.unwindFlag} ${args} --output-format terse`;
-		}
-		const kaniOutput = await createFailedDiffMessage(harnessCommand);
-		return kaniOutput;
+	if (args === undefined || NaN) {
```
Comment (adpaco-aws) [inline src/model/kaniCommandCreate.ts:78]: Does this condition need to be `args === undefined || args === NaN`?

File: src/model/kaniRunner.ts [hunk 3]
```diff
@@ -62,7 +62,7 @@ export async function runKaniCommand(
 	// Get cargo command and args for the command to be executed
 	const command = commmandSplit.commandPath;
 	const args = commmandSplit.args;
-	let kaniBinaryPath = '';
+	const kaniBinaryPath = '';
```
Comment (adpaco-aws) [inline src/model/kaniRunner.ts:65]: I think you can remove this declaration altogether. The variable seems to be shadowed a few lines below.

**Review by adpaco-aws — COMMENTED** [review]
It makes sense for the extension to only use `cargo kani` commands, but I have a few questions.

File: src/model/kaniRunner.ts (hunk 3 above)
Comment (jaisnan) [reply src/model/kaniRunner.ts:65]: Thanks for pointing that out!

File: src/model/kaniCommandCreate.ts (hunk 2 above)
Comment (jaisnan) [reply src/model/kaniCommandCreate.ts:78]: Yes, since args is an optional number (although this will be modified by the tree-sitter anyway), which means it can be a number or undefined. But even if it's passing a number, typescript encodes errors as numbers as a NaN value (not a number). So we are checking for cases where either there is no args or there is an args but there is an incorrect or erroneous value.

**Review by adpaco-aws — APPROVED** [review]
Thanks, @jaisnan !
